packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -1,216 +1,460 @@
 
-    // Mapping of lender address => market ID => lending token => commitment
-    mapping(address => mapping(uint256 => mapping(address => Commitment)))
-        public lenderMarketCommitments;
+    uint256 commitmentCount;
+
+    mapping(uint256 => EnumerableSetUpgradeable.AddressSet)
+        internal commitmentBorrowersList;
 
     /**
-     * @notice This event is emitted when a lender's commitment is submitted.
+     * @notice This event is emitted when a lender's commitment is created.
      * @param lender The address of the lender.
      * @param marketId The Id of the market the commitment applies to.
      * @param lendingToken The address of the asset being committed.
      * @param tokenAmount The amount of the asset being committed.
      */
-    event UpdatedCommitment(
-        address indexed lender,
-        uint256 indexed marketId,
-        address indexed lendingToken,
+    event CreatedCommitment(
+        uint256 indexed commitmentId,
+        address lender,
+        uint256 marketId,
+        address lendingToken,
         uint256 tokenAmount
     );
 
     /**
-     * @notice This event is emitted when a lender's commitment has been removed.
+     * @notice This event is emitted when a lender's commitment is updated.
      * @param lender The address of the lender.
-     * @param marketId The Id of the market the commitment removal applies to.
-     * @param lendingToken The address of the asset the commitment removal applies to.
+     * @param marketId The Id of the market the commitment applies to.
+     * @param lendingToken The address of the asset being committed.
+     * @param tokenAmount The amount of the asset being committed.
      */
-    event DeletedCommitment(
-        address indexed lender,
-        uint256 indexed marketId,
-        address indexed lendingToken
+    event UpdatedCommitment(
+        uint256 indexed commitmentId,
+        address lender,
+        uint256 marketId,
+        address lendingToken,
+        uint256 tokenAmount
     );
 
+    /**
+     * @notice This event is emitted when a lender's commitment has been deleted.
+     * @param commitmentId The id of the commitment that was deleted.
+     */
+    event DeletedCommitment(uint256 indexed commitmentId);
+
     /**
      * @notice This event is emitted when a lender's commitment is exercised for a loan.
-     * @param lender The address of the lender.
+     * @param borrower The address of the borrower.
      * @param marketId The Id of the market the commitment applies to.
      * @param lendingToken The address of the asset being committed.
      * @param tokenAmount The amount of the asset being committed.
      * @param bidId The bid id for the loan from TellerV2.
      */
     event ExercisedCommitment(
-        address indexed lender,
-        uint256 indexed marketId,
-        address indexed lendingToken,
+        uint256 indexed commitmentId,
+        address borrower,
```

Comment:
we should not need to emit the `borrower` either as we can infer the borrower's address from the `bidId` in the event
```suggestion
```

Reply:
i think it makes the subgraph querying simpler if we leave emit borrower here though ..  we cannot infer the borrowers address from bidId unless you mean an event on another contract ?